src/sysroot.rs

```diff
@@ -231,9 +231,11 @@ pub fn update(
     verbose: bool,
 ) -> Result<()> {
     let ctoml = cargo::toml(root)?;
-    let xtoml = xargo::toml(root)?;
+    let (xtoml_parent, xtoml) = xargo::toml(root)?;
 
-    let blueprint = Blueprint::from(xtoml.as_ref(), cmode.triple(), root, &src)?;
+    let base_path = xtoml_parent.unwrap_or_else(|| root.path().to_path_buf());
```

Comment:
Seems a bit wasteful to make a copy here with `to_path_buf` if we only need an `&Path` below anyway. Can that be avoided?